### PR TITLE
init: Fix F401 Linter Warnings by Replacing Wildcard Imports with Explicit Imports in   __init__.py

### PR DESCRIPTION
**Pull request by arohanajit**
Addresses multiple `F401` linter warnings in the  `__init__.py`  file by replacing wildcard imports with explicit imports and defining the  `__all__` variable to specify the public API of the module. Replaced Wildcard Imports: Removed `from .data import *` and explicitly imported the necessary functions and classes. Also updated `.flake8` since no `F403` warnings were found

### CODE REVIEW

**Comment by echoix** [comment]
It's me that cancelled the remaining jobs to that were making #4646 waiting to be merged. That PR will make the builds work again, and is required to have the required macOS check be successful, so this PR would have been updated minutes after that

File: python/grass/script/__init__.py [hunk 1]
```diff
@@ -1,10 +1,222 @@
...
+    "set_path",
+    "set_raise_on_error",
+    "split",
+    "start_command",
+    "tempdir",
+    "tempfile",
+    "tempname",
+    "text_to_string",
+    "try_remove",
+    "try_remove",
```
Comment (petrasovaa) [inline python/grass/script/__init__.py:204]: You have it twice there

File: python/grass/script/__init__.py [hunk 2]
```diff
@@ -1,10 +1,222 @@
...
+    vector_history,
+    vector_info_topo,
+    vector_info,
+    vector_db_select,
+    vector_what,
+)
 from . import setup  # noqa: F401
+
+__all__ = [
+    "PIPE",
```
Comment (petrasovaa) [inline python/grass/script/__init__.py:112]: The ordering looks like it mixes alphabetical ordering and order by the modules, I am not sure if there is any convention, I would prefer ordering by the modules, it's easier to check.

Comment (arohanajit) [reply python/grass/script/__init__.py:112]: I had initially ordered it by modules itself, however precommit reorders it every time i commit it to alphabetical style

Comment (petrasovaa) [reply python/grass/script/__init__.py:112]: ok, then that's fine

Comment (petrasovaa) [reply python/grass/script/__init__.py:112]: just remove the comments then, they are misleading

**Review by petrasovaa — CHANGES_REQUESTED** [review]
You have several duplications in __all__...

**Review by petrasovaa — CHANGES_REQUESTED** [review]
You need to run isort and the CI fails because there is missing Popen class from core.py

File: python/grass/script/__init__.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
 """Python interface to launch GRASS GIS modules in scripts
 """
 
+from . import setup  # noqa: F401
```
Comment (petrasovaa) [inline python/grass/script/__init__.py:4]: Why this?

Comment (arohanajit) [reply python/grass/script/__init__.py:4]: This was already there so I didn't change it. I'm assuming for some dependencies since it's not used directly

Comment (echoix) [reply python/grass/script/__init__.py:4]: It was me like a couple weeks ago. Ruff, in preview mode, didn't do the right suggestion, or had a suggestion loop of some sort. Or maybe the fix was worse with another tool, like pyright, as the file is named setup.py, and it's not a setuptools setup.py file.


So, you might want to try again, it wasn't a big bug.

**Review by petrasovaa — CHANGES_REQUESTED** [review]
The CI fails:
```
  File "scripts/t.vect.import", line 113, in main
    tgis.import_stds(
  File "etc/python/grass/temporal/stds_import.py", line 250, in import_stds
    old_state = gs.raise_on_error
AttributeError: module 'grass.script' has no attribute 'raise_on_error'. Did you mean: 'get_raise_on_error'?
```
because there is missing raise_on_error from core.py. But I would say in this case it should use the getter anyway:
```
old_state = gs.get_raise_on_error()
```
So the variable probably should not be exposed.

**Comment by arohanajit** [comment]
> The CI fails:
> 
> ```
>   File "scripts/t.vect.import", line 113, in main
>     tgis.import_stds(
>   File "etc/python/grass/temporal/stds_import.py", line 250, in import_stds
>     old_state = gs.raise_on_error
> AttributeError: module 'grass.script' has no attribute 'raise_on_error'. Did you mean: 'get_raise_on_error'?
> ```
> 
> because there is missing raise_on_error from core.py. But I would say in this case it should use the getter anyway:
> 
> ```
> old_state = gs.get_raise_on_error()
> ```
> 
> So the variable probably should not be exposed.

So in this case CI will probably always fail unless we expose 'raise_on_error()` which we shouldn't. Is there another way to manage this?

**Comment by petrasovaa** [comment]
> > The CI fails:
> > ```
> >   File "scripts/t.vect.import", line 113, in main
> >     tgis.import_stds(
> >   File "etc/python/grass/temporal/stds_import.py", line 250, in import_stds
> >     old_state = gs.raise_on_error
> > AttributeError: module 'grass.script' has no attribute 'raise_on_error'. Did you mean: 'get_raise_on_error'?
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > because there is missing raise_on_error from core.py. But I would say in this case it should use the getter anyway:
> > ```
> > old_state = gs.get_raise_on_error()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So the variable probably should not be exposed.
> 
> So in this case CI will probably always fail unless we expose 'raise_on_error()` which we shouldn't. Is there another way to manage this?

I was suggesting to do the change in stds_import.py

**Review by petrasovaa — DISMISSED** [review]

**Comment by echoix** [comment]
The .flake8 changes mentioned in the PR description were lost in one of the commits. I tried cleaning it back locally, iterating by using `pre-commit run -a flake8` between each change. So the __init__.py exclusions can be limited to python/grass/temporal.

**Review by echoix — APPROVED** [review]

**Pull request merged**